src/pages/collators/CollatorRewards.tsx

```diff
@@ -0,0 +1,160 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import RewardsIcon from '../../assets/collators-rewards-icon';
+import StakedIcon from '../../assets/collators-staked-icon';
+import { nativeToFormat } from '../../helpers/parseNumbers';
+import { UserStaking } from './columns';
+import { useNodeInfoState } from '../../NodeInfoProvider';
+import { useGlobalState } from '../../GlobalStateProvider';
+import { getAddressForFormat } from '../../helpers/addressFormatter';
+import { ParachainStakingCandidate, useStakingPallet } from '../../hooks/staking/staking';
+import ClaimRewardsDialog from './dialogs/ClaimRewardsDialog';
+import { toast } from 'react-toastify';
+import { getErrors, getEventBySectionAndMethod } from '../../helpers/substrate';
+import { Button } from 'react-daisyui';
+
+function CollatorRewards() {
+  const [userAvailableBalance, setUserAvailableBalance] = useState<string>('0.00');
+  const [userStaking, setUserStaking] = useState<UserStaking>();
+  const [claimDialogOpen, setClaimDialogOpen] = useState<boolean>(false);
+  const [submissionPending, setSubmissionPending] = useState(false);
+
+  const { api, tokenSymbol, ss58Format } = useNodeInfoState().state;
+  const { walletAccount } = useGlobalState();
+  const { candidates, estimatedRewards, createUpdateDelegatorRewardsExtrinsic } = useStakingPallet();
+
+  const userAccountAddress = useMemo(() => {
+    return walletAccount && ss58Format ? getAddressForFormat(walletAccount?.address, ss58Format) : '';
+  }, [walletAccount, ss58Format]);
+
+  useMemo(() => {
+    setUserStaking(undefined);
+    return candidates?.forEach((candidate) => {
+      const isDelegator = candidate.delegators.find((delegator) => delegator.owner === userAccountAddress);
+      if (isDelegator) {
+        setUserStaking({
+          candidateId: candidate.id,
+          amount: isDelegator.amount,
+        });
+      }
+    });
+  }, [candidates, userAccountAddress, setUserStaking]);
+
+  useEffect(() => {
+    const fetchAvailableBalance = async () => {
+      if (!api || !walletAccount) {
+        return '0';
+      }
+      const { data: balance } = await api.query.system.account(walletAccount?.address);
+      return balance.free.sub(balance.miscFrozen).toString();
+    };
+
+    fetchAvailableBalance().then((balance) => setUserAvailableBalance(balance));
+  }, [api, walletAccount]);
+
+  const updateRewardsExtrinsic = useMemo(() => {
+    if (!api) {
+      return undefined;
+    }
+
+    return createUpdateDelegatorRewardsExtrinsic();
```

Comment:
Can we also update the collator rewards in this function? Maybe we can batch the `incrementDelegatorRewards()` and `incrementCollatorRewards()` extrinsics to one transaction so that the user doesn't have to sign twice. But updating both would be preferrable.

Reply:
I'll take that, will modify it :+1: